metadata/trustedmetadata/trustedmetadata_test.go

```diff
@@ -22,48 +22,56 @@ import (
 	"github.com/rdimitrov/go-tuf-metadata/testutils/testutils"
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
 	"github.com/sigstore/sigstore/pkg/signature"
-	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/exp/maps"
 )
 
 var allRoles map[string][]byte
 
 func setAllRolesBytes(path string) {
+	log := metadata.GetLogger()
```

Comment:
Should we set a logger for the tests? If I get it right, we default to a nil logger so these will not log, not that tests need logs that much, but still?

Another thing - do you think it's better if we should explicitly fail/assert on values instead of having `if` statements for error handling on things which did not executed as expected? My assumption is tests should be deterministic so their workflow shouldn't change.

Reply:
In my opinion for the only logger for tests that should be used is `t.Log` as it is coupled with the `go test` tool. So I agree with you, that we shouldn't set a logger in the tests, I merely kept the old behavior in this case.
And yes I fully agree with you on the testing part that we should always fail if the state of a test was unexpected. But this was out of scope for the PR so I left it as is for now. (Especially given how large the test files are)

Reply:
Nice, I'll create a separate issue about removing such ambiguity in the tests 👍